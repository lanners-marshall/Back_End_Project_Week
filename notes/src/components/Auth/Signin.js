import React from 'react';
import axios from 'axios';
import { Link } from 'react-router-dom';
import { Contain, MainH1, FlexForm, BTN, SubmitBtn, BTNDiv, LogErr } from './css';


class Signin extends React.Component {
	constructor(props){
		super(props);
		this.state = {
			username: '',
			password: '',
			error: localStorage.getItem("error"),
			failedLogin: '',
		};
	}

	handleChange = event => {
 	   this.setState({[event.target.name]: event.target.value})
 	}

 	signin = event => {
 		event.preventDefault();
 		//console.log('its working')
 		axios.post('http://localhost:5555/users/login', this.state)
 		.then(response => {
 			//console.log(response.data)
 			localStorage.setItem('jwt', response.data.token);
 			localStorage.setItem('loggedInAs', this.state.username);
 			this.setState({
 				failedLogin: ''
 			})
 			this.props.history.push('/notes')
 		})
 		.catch(error => {
 			console.log(error.response.data)
 			this.setState({
 				failedLogin: error.response.data.msg
 			})
 		})
 	}

 	clear = event => {
 		localStorage.removeItem('error');
 	}

	render() {
		//console.log(this.state)
		return (
			<div>
				<LogErr>{this.state.error}</LogErr>
				<Contain>
					<MainH1>Log In</MainH1>
					<FlexForm>
						<input
							type="text"
							placeholder='username'
							onChange={this.handleChange}
							name="username"
							value={this.state.username}
						/>
						<input
							type="password"
							placeholder='password'
							onChange={this.handleChange}
							name="password"
							value={this.state.password}
						/>
						<SubmitBtn onClick={this.signin}>Signin</SubmitBtn>
					</FlexForm>

				</Contain>
				<Link to='signup'><BTNDiv onClick={this.clear}><BTN>No account Sign up here!</BTN></BTNDiv></Link>
				<LogErr>{this.state.failedLogin}</LogErr>
			</div>
		)
	}
}

export default Signin;