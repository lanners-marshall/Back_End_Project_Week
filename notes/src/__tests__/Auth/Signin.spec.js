import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter, Route } from 'react-router-dom';

import Signin from '../../components/Auth/Signin';
import renderer from 'react-test-renderer';

describe('<Signin />', () => {
  it('should match snapshot', () => {
	  const wrapper = shallow(
	  	<MemoryRouter initialEntries={['/signin']}>
	      <Signin />
	    </MemoryRouter>
  	)
  	const tree = renderer.create(wrapper).toJSON()
  	expect(tree).toMatchSnapshot()
	})
});