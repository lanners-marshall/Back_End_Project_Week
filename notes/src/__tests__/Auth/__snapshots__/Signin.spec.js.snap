// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signin /> should match snapshot 1`] = `
<div>
  <p
    className="sc-gzVnrw jvxosd"
  />
  <div
    className="sc-bdVaJa hZjtPt"
  >
    <h1
      className="sc-bwzfXH grlnmy"
    >
      Log In
    </h1>
    <form
      className="sc-htpNat fvVwSE"
    >
      <input
        name="username"
        onChange={[Function]}
        placeholder="username"
        type="text"
        value=""
      />
      <input
        name="password"
        onChange={[Function]}
        placeholder="password"
        type="password"
        value=""
      />
      <button
        className="sc-ifAKCX iUOQTm"
        onClick={[Function]}
      >
        Signin
      </button>
    </form>
  </div>
  <a
    href="/signup"
    onClick={[Function]}
  >
    <div
      className="sc-EHOje hqEBXS"
      onClick={[Function]}
    >
      <button
        className="sc-bxivhb eMpRCh"
      >
        No account Sign up here!
      </button>
    </div>
  </a>
  <p
    className="sc-gzVnrw jvxosd"
  >
    
  </p>
</div>
`;
